packaging/suse/portusctl/lib/configurator.rb

```diff
@@ -0,0 +1,186 @@
+class Configurator
+  def initialize(options)
+    @options         = options
+    @secret_key_base = SecureRandom.hex(64)
+    @portus_password = SecureRandom.hex(64)
+  end
+
+  # Performs the folloing operation:
+  #   * create vhost configuration for Portus
+  #   * enable mod_passenger
+  #   * enable mod_ssl only if secure option is enabled
+  def apache
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'apache2_portus.conf.erb')
+
+    conf  = ERB.new(File.read(template), nil, '<>').result(binding)
+
+    File.open("/etc/apache2/vhosts.d/portus.conf", "w") do |file|
+      file.write(conf)
+    end
+
+    Runner.exec("a2enmod", ["ssl"]) if @options["secure"]
+    Runner.exec("a2enmod", ["passenger"])
+  end
+
+  # Performs the following operations:
+  #   * create the ssl certificates
+  #   * copy the cerificates to the right locations
+  def ssl
+    unless File.exist?("/etc/apache2/ssl.key/#{HOSTNAME}-server-key")
+      puts "Generating private key and certificate"
+      puts ""
+      puts "***********************************************************************************************************************************"
+      puts "If you want to use your own private key and certificates, upload them to"
+      puts "  * /etc/apache2/ssl.key/#{HOSTNAME}-server.key"
+      puts "  * /etc/apache2/ssl.crt/#{HOSTNAME}-server.crt"
+      puts "  * /etc/apache2/ssl.crt/#{HOSTNAME}-ca.crt"
+      puts "and then re-run this script"
+      puts "***********************************************************************************************************************************"
+
+      args = [
+        "-C", HOSTNAME,
+        "-n", HOSTNAME,
+        "-o", @options["ssl-organization"],
+        "-u", @options["ssl-organization-unit"],
+        "-e", @options["ssl-email"],
+        "-c", @options["ssl-country"],
+        "-l", @options["ssl-city"],
+        "-s", @options["ssl-state"]
+      ]
+      Runner.exec("gensslcert", args)
+    end
+
+    FileUtils.chown("wwwrun", "www", "/etc/apache2/ssl.key")
+    FileUtils.chmod(750,"/etc/apache2/ssl.key")
+
+    FileUtils.chown("wwwrun", "www", "/etc/apache2/ssl.key/#{HOSTNAME}-server.key")
+    FileUtils.chmod(440,"/etc/apache2/ssl.key/#{HOSTNAME}-server.key")
+
+    # Create key used by Portus to sign the JWT tokens
+    FileUtils.ln_sf(
+      "/etc/apache2/ssl.key/#{HOSTNAME}-server.key",
+      File.join("/srv/Portus/config", "server.key"))
+
+    FileUtils.cp(
+      "/etc/apache2/ssl.crt/#{HOSTNAME}-ca.crt",
+      "/srv/www/htdocs")
+    FileUtils.chmod(755,"/srv/www/htdocs/#{HOSTNAME}-ca.crt")
+    FileUtils.cp(
+      "/etc/apache2/ssl.crt/#{HOSTNAME}-ca.crt",
+      "/etc/pki/trust/anchors")
+    Runner.exec("update-ca-certificates")
+  end
+
+  # Creates the database.yml file required by Rails
+  def database_config
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'database.yml.erb')
+
```

Comment:
chmod/chown so that only root can write and only wwwrun can read.
TODO: database.yml should not be owned by the package because we are modifying/creating it